spec/unread/readable_spec.rb

```diff
@@ -266,6 +266,17 @@
       expect(@email2.unread?(@reader)).to be_falsey
     end
 
+    it "should mark the rest as read when the first record is not unique" do
+      Email.mark_as_read! [ @email1 ], for: @reader
+
+      allow(@email1).to receive_message_chain("read_marks.build").and_return(@email1.read_marks.build)
+      allow(@email1).to receive_message_chain("read_marks.where").and_return([])
+
+      Email.mark_as_read! [ @email1, @email2 ], for: @reader
+      expect(@email1.unread?(@reader)).to be_falsey
+      expect(@email2.unread?(@reader)).to be_falsey
+    end
```

Comment:
Please wrap the whole example into a `expect { ... }.to change(ReadMark, :count).by(2)`. This makes sure that the example fails if the database index is not unique.